ui/app/AppLayouts/Wallet/services/dapps/DappsConnectorSDK.qml

```diff
@@ -148,9 +148,22 @@ WalletConnectSDKBase {
                 }
                 address = event.params.request.params[0]
             }
-            return SQUtils.ModelUtils.getFirstModelEntryIf(root.wcService.validAccounts, (account) => {
+            const account = SQUtils.ModelUtils.getFirstModelEntryIf(root.wcService.validAccounts, (account) => {
                 return account.address.toLowerCase() === address.toLowerCase();
             })
+
+            if (!account) {
+                return null
+            }
+
+            // deep copy to avoid operations on the original object
+            try {
+                return JSON.parse(JSON.stringify(account))
+            }
```

Comment:
nit: Should the catch go on the same line as `}` seems like you are consistent for other changes, so feel free to ignore.